tests/pypi/pkg_aliases/pkg_aliases_test.bzl

```diff
@@ -0,0 +1,217 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""pkg_aliases tests"""
+
+load("@rules_testing//lib:test_suite.bzl", "test_suite")
+load(
+    "//python/private/pypi:pkg_aliases.bzl",
+    "pkg_aliases",
+)  # buildifier: disable=bzl-visibility
+
+_tests = []
+
+def _test_legacy_aliases(env):
+    actual = []
+    pkg_aliases(
+        name = "foo",
+        actual = "repo",
+        native = struct(
+            alias = lambda **kwargs: actual.append(kwargs),
+        ),
+        extra_aliases = ["my_special"],
+    )
+
+    # buildifier: disable=unsorted-dict-items
+    want = [
+        {
+            "name": "foo",
+            "actual": ":pkg",
+        },
+        {
+            "name": "pkg",
+            "actual": "@repo//:pkg",
+        },
+        {
+            "name": "whl",
+            "actual": "@repo//:whl",
+        },
+        {
+            "name": "data",
+            "actual": "@repo//:data",
+        },
+        {
+            "name": "dist_info",
+            "actual": "@repo//:dist_info",
+        },
+        {
+            "name": "my_special",
+            "actual": "@repo//:my_special",
+        },
+    ]
+
+    env.expect.that_collection(actual).contains_exactly(want)
+
+_tests.append(_test_legacy_aliases)
+
+def _test_config_setting_aliases(env):
+    # Use this function as it is used in pip_repository
+    actual = []
+    actual_no_match_error = []
+
+    def mock_select(value, no_match_error = None):
+        actual_no_match_error.append(no_match_error)
+        env.expect.that_str(no_match_error).equals("""\
```

Comment:
Oh ouch, this is brittle. I gotta add some StrSubject methods to make partial matching easier. FYI: .contains() can be used for less exact matching